app/src/hooks/useParaspellApi.ts

```diff
@@ -260,6 +269,50 @@ const useParaspellApi = () => {
       status: `Arriving at ${transferToStore.destChain.name}`,
       finalizedAt: new Date(),
     })
+
+    monitorBatchSwap(transferToStore, onchainEvents)
+    handleLocalSwapStorage(transferToStore)
+  }
+
+  const isBatchCompleted = (onchainEvents: PjsEvents | PapiEvents) => {
+    return !!('isBatchCompleted' in onchainEvents && onchainEvents.isBatchCompleted === true)
+  }
+
+  const monitorBatchSwap = (transfer: StoredTransfer, eventsData: PjsEvents | PapiEvents) => {
+    // Non local swap are handled with the BatchAll extinsic from utility pallet
```

Comment:
To be more precise with the language. swaps are always local. 
```suggestion
    // swap + transer are handled with the BatchAll extinsic from utility pallet
```